src/bloom.cpp

```diff
@@ -242,9 +264,20 @@ bool CRollingBloomFilter::contains(const std::vector<unsigned char>& vKey) const
     return b1.contains(vKey);
 }
 
-void CRollingBloomFilter::clear()
+bool CRollingBloomFilter::contains(const uint256& hash) const
 {
-    b1.clear();
-    b2.clear();
+    if (nInsertions < nBloomSize / 2) {
+        return b2.contains(hash);
+    }
+    return b1.contains(hash);
+}
+
+void CRollingBloomFilter::reset(unsigned int nNewTweak)
+{
+    if (!nNewTweak)
+        nNewTweak = GetRand(std::numeric_limits<unsigned int>::max());
```

Comment:
On Mon, Jul 27, 2015 at 04:35:45AM -0700, Wladimir J. van der Laan wrote:
> >  {
> > -    b1.clear();
> > -    b2.clear();
> > +    if (nInsertions < nBloomSize / 2) {
> > +        return b2.contains(hash);
> > +    }
> > +    return b1.contains(hash);
> > +}
> > +
> > +void CRollingBloomFilter::reset(unsigned int nNewTweak)
> > +{
> > +    if (!nNewTweak)
> > +        nNewTweak = GetRand(std::numeric_limits<unsigned int>::max());
> 
> I don't like using 0 as a special marker value here - wouldn't 0 be a perfectly good tweak value?
> Maybe add an explicit method `resetWithRandomTweak()`?

Well, I'm expecting people to always be using the random tweak provided
except in testing.

Why would you need a specific tweak given the internal use-case?